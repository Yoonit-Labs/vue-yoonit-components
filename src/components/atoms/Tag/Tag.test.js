import { shallowMount } from '@vue/test-utils'
import PropsConfig from './Tag.config'
import YooTag from './Tag.vue'

const classBlock = 'yoo-tag'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooTag, {
    slots: { default: SlotText },
    propsData: {
      tags: [
        {
          text: 'Admin',
          status: true
        },
        {
          text: 'Funcionários',
          status: false
        },
        {
          text: 'Segurança',
          status: false
        },
        {
          text: 'Manutenção'
        }
      ]
    }
  })
}

describe('YooTag Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', async () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.classes(`${classBlock}`)).toBe(true)
    expect(wrapper.find(`${classBlock}__item`).exists()).toBe(true)
  })

  describe('Props', () => {
    describe('fill', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.fill.options.includes(YooTag.props.fill.default)).toBe(true)
      })
      PropsConfig.fill.options.forEach(fill => {
        it(`Has class .${classBlock}__fill--${fill}`, async () => {
          await wrapper.setProps({ fill })
          expect(wrapper.find(`.${classBlock}__fill--${fill}`).exists()).toBe(true)
        })
      })
    })
  }) // describe Props
})
