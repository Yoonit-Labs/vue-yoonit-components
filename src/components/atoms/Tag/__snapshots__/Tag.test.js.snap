// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooTag Component Matches Snapshot 1`] = `
<yooflexlayout-stub flexdirection="row" flexwrap="wrap" justifycontent="flex-start" alignitems="stretch" aligncontent="stretch" gap="4px" class="yoo-tag">
  <p class="yoo-tag__item yoo-tag__item--on yoo-tag__fill--primary">Admin</p>
  <p class="yoo-tag__item yoo-tag__fill--primary">Funcionários</p>
  <p class="yoo-tag__item yoo-tag__fill--primary">Segurança</p>
  <p class="yoo-tag__item yoo-tag__fill--primary">Manutenção</p>
</yooflexlayout-stub>
`;
